### ISSUE TITLE
How to train on my own dataset in version 2 ?

### ISSUE THREAD

**Issue by markunys** [hukkelas/DeepPrivacy#29]
Hello, thank you for this great work.

I am trying to train retinanet512 model on my own dataset, but there aren't examples of usage of train.py in version 2.
How do I train on my own dataset in version 2 ?

**Comment by hukkelas** [comment]
Hi! 

If the dataset is not already implemented (we've implemented places2, celeba-hq and FDF), you need to define your own Dataset class, and a config file for this dataset.

All datasets are build on top of the [CustomDataset](https://github.com/hukkelas/DeepPrivacy/blob/3b2d7e5049e9261d1e517a20395ff04d8090ddb7/deep_privacy/dataset/custom.py#L6) class, which implements how to read images (this class assumes that the dataset follows the directory structure: data/dataset_name/images/imsize, for example data/fdf/images/128).

Specifically, you need to edit `get_mask` (mask is the area to remove) and `_load_impaths` to read your images (see [CelebA-HQ](https://github.com/hukkelas/DeepPrivacy/blob/3b2d7e5049e9261d1e517a20395ff04d8090ddb7/deep_privacy/dataset/celebaHQ.py#L7) for example). Then the dataset returns the image and mask by default. If you need to train with landmarks/keypoints, you need  to edit a bit more, see how we implemented the [FDF dataset](https://github.com/hukkelas/DeepPrivacy/blob/3b2d7e5049e9261d1e517a20395ff04d8090ddb7/deep_privacy/dataset/fdf.py#L15).

When all is done, **you have to define a config file**. For example, we modified the CelebA-HQ dataset as seen [here](https://github.com/hukkelas/DeepPrivacy/blob/master/configs/celebA-HQ.py), or for the FDF dataset with landmarks the pipeline is somewhat more complicated, as seen [here](https://github.com/hukkelas/DeepPrivacy/blob/master/configs/fdf/base.py).

When all is done, you can test out your pipeline by checking that the images looks correct with (change config path to your config):
```
python -m deep_privacy.dataset.build configs/celebA-HQ.py
``` 

Hope this helps!

**Comment by markunys** [comment]
Hello.

I solved this problem. Thank you.

I have another question.  
I tried to use your pre-trained model, 'fdf128_retinanet512'(retinanet512.ckpt), as initial checkpoint, when I trained my model.
but, retinanet512.ckpt has only 'running_average_generator'.
Are other checkpoints('D', 'G', 'optimizer', etc) available ?
